src/main/java/core/basesyntax/strategy/handler/DecrementHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public abstract class DecrementHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        fruit = Storage.fruits.get(fruit.getName());
```

Comment:
Directly assigning a new value to the parameter 'fruit' can be misleading. If you intend to update the fruit in the storage, consider using a different local variable to hold the reference from the storage.